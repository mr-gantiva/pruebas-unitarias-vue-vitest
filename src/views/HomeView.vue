<script setup>
import { RouterLink, RouterView } from 'vue-router'
import Posts from '@/components/Posts.vue';
</script>

<template>
    <main>
        <!-- <TheWelcome /> -->

        <nav>
            <RouterLink to="/about">About</RouterLink>
            <RouterLink to="/contact">Contact</RouterLink>
        </nav>

        <RouterView />

        <Posts />
    </main>
</template>
