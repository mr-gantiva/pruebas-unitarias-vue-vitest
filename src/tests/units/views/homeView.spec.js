import { shallowMount } from '@vue/test-utils'
import HomeView from '@/views/HomeView.vue'
import { describe, test, expect } from 'vitest'

describe('Componente HomeView.vue', () => {
  test('Snapshot de HomeView.vue', () => {
    // Selección del sujeto de pruebas
    const wrapper = shallowMount(HomeView)
    // Validación de match con el snapshot
    expect(wrapper.html()).toMatchSnapshot()
  })
})
