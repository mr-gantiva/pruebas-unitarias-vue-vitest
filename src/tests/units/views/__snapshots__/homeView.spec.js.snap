// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Componente HomeView.vue > Snapshot de HomeView.vue 1`] = `
"<main>
  <!-- <TheWelcome /> -->
  <nav>
    <router-link-stub to="/about" replace="false" custom="false" ariacurrentvalue="page"></router-link-stub>
    <router-link-stub to="/contact" replace="false" custom="false" ariacurrentvalue="page"></router-link-stub>
  </nav>
  <router-view-stub name="default"></router-view-stub>
  <posts-stub></posts-stub>
</main>"
`;

exports[`Componente HomeView.vue > Validaci√≥n de match con el snapshot 1`] = `
"<main>
  <!-- <TheWelcome /> -->
  <nav>
    <router-link-stub to="/about" replace="false" custom="false" ariacurrentvalue="page"></router-link-stub>
    <router-link-stub to="/contact" replace="false" custom="false" ariacurrentvalue="page"></router-link-stub>
  </nav>
  <router-view-stub name="default"></router-view-stub>
  <posts-stub></posts-stub>
</main>"
`;
